sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NormalizeFloatingNumbers.scala

```diff
@@ -123,7 +123,8 @@ object NormalizeFloatingNumbers extends Rule[LogicalPlan] {
       val fields = expr.dataType.asInstanceOf[StructType].fields.indices.map { i =>
         normalize(GetStructField(expr, i))
       }
-      CreateStruct(fields)
```

Comment:
Do you know why this works before?

Reply:
`CreateStruct` re-creates the struct when `NormalizeFloatingNumbers` is run. The original `expr` is not used. But now we wrap `expr` in `IsNull`.

Reply:
ah, got it!

This reminds me that we can optimize the code a bit:  Instead of only matching `CreateNamedStruct`, we can have a whitelist to apply the normalization to the children. For example, `If(..., CreateNamedStruct(...))`, we should apply normalization to the inner `CreateNamedStruct`.

We can do this later.

Reply:
Sure.

Reply:
For `If(..., CreateNamedStruct(...))`, it is struct type so already covered by the `StructType` case. So the normalization is already applied to the inner `CreateNamedStruct`.

Reply:
yea I know, but I mean we can do better. We can directly apply the normalization to the `children` in `If(..., CreateNamedStruct(children))`, instead of treating the outer `If` as a black box and apply normalization to it.